tests/api_tests/test_search.py

```
@@ -0,0 +1,94 @@
+
+    def test_search_partition(self, api_client, indexed_partition):
+        """Test searching within a partition."""
+        response = api_client.get(
+            f"/search/partition/{indexed_partition}",
+            params={"text": "artificial intelligence", "top_k": 5}
+        )
+        assert response.status_code == 200
+        data = response.json()
+        assert "documents" in data
+
+    def test_search_multiple_partitions(self, api_client, indexed_partition):
```

Ruff (0.14.10)

Unused method argument: `indexed_partition`

(ARG002)